src/insert-extension.js

```diff
@@ -83,33 +79,54 @@ function isAmpExtensionScriptRequired(win, extension) {
 };
 
 /**
- * Calculate script url for amp-ad.
+ * Calculate base script url for any js files.
  * @visibleForTesting
- * @param {string} path Location path of the window
- * @param {string} extension
- * @param {boolean=} isTest
- * @param {boolean=} isUsingCompiledJs
+ * @param {!Window} win
  * @return {string}
  */
-export function calculateExtensionScriptUrl(path, extension, isTest,
-  isUsingCompiledJs) {
-  if (getMode().localDev) {
+export function calculateScriptBase(win) {
+  const isLocalDev = getMode(win).localDev;
+  const isTest = getMode(win).test;
+  if (isLocalDev) {
     if (isTest) {
-      if (isUsingCompiledJs) {
-        return `/base/dist/v0/${extension}-0.1.js`;
-      }
-      return `/base/dist/v0/${extension}-0.1.max.js`;
+      return '/base/dist';
     }
-    if (path.indexOf('.max') >= 0) {
-      return `http://localhost:8000/dist/v0/${extension}-0.1.max.js`;
+    const path = win.location.pathname;
+    if (path.indexOf('.min') >= 0 || path.indexOf('.max') >= 0) {
+      return 'http://localhost:8000/dist';
     }
-    if (path.indexOf('.min') >= 0) {
-      return `http://localhost:8000/dist/v0/${extension}-0.1.js`;
+    return 'https://cdn.ampproject.org';
+  }
+
+  const domain = 'https://cdn.ampproject.org';
+  const version = getMode(win).version;
+  const folderPath = version == '$internalRuntimeVersion$' ?
+      '' : `rtv/${version}`;
+  return `${domain}${folderPath}`;
+}
+
+/**
+ * Calculate script url for amp-ad.
+ * @visibleForTesting
+ * @param {!Window} win
+ * @param {string} extension
+ * @return {string}
+ */
+export function calculateExtensionScriptUrl(win, extension) {
+  const isLocalDev = getMode(win).localDev;
+  const isTest = getMode(win).test;
```

Comment:
You're currently working on this method? I can defer to what you've written instead.